dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/DefaultProgramIndicatorService.java

```diff
@@ -213,21 +214,7 @@ public void validate( String expression, Class<?> clazz, Map<String, String> ite
 
     @Override
     @Transactional(readOnly = true)

```

Comment:
I wonder if we can get rid of the `@Transactional` annotation at all here, since, it looks to me the code doesn't not hit the database at all. It's better to avoid starting TX if we don't need them.

Reply:
ProgramSqlGenerator.visitItem calls dataElementService.getDataElement and attributeService.getTrackedEntityAttribute to get DataElement and Attribute types, to know whether to replace nulls with 0 or ''.